### PR TITLE
(feat):fix  prompt system

### PR DESCRIPTION
**Pull request by jettpihu**
(feat):fix  prompt system

### CODE REVIEW

File: query_analyzer.py [hunk 1]
```diff
@@ -12,55 +12,71 @@
...
+- speech to text/transcription -> automatic-speech-recognition
+- text spoken aloud -> text-to-speech
+- music/sound generation from text -> text-to-audio
+- video generation from text -> text-to-video
+
+Examples:
+- "Summarize: AI is changing the world..." -> {{"task":"summarization","search_query":"summarization","inference_input":"AI is changing the world...","needs_inference":true}}
+- "Translate hello to French" -> {{"task":"translation","search_query":"translation english french","inference_input":"hello","needs_inference":true}}
+- "What is the sentiment of: I love this!" -> {{"task":"text-classification","search_query":"sentiment analysis","inference_input":"I love this!","needs_inference":true}}
+- "Find named entities in: Elon Musk founded SpaceX in California" -> {{"task":"token-classification","search_query":"named entity recognition","inference_input":"Elon Musk founded SpaceX in California","needs_inference":true}}
```
Comment (gautammanak1) [inline query_analyzer.py]: The fill-mask example here appears to have a malformed mask token `"The capital of France is </think>"`. This could confuse the model. Consider using `[MASK]` or `<mask>` instead, matching the keywords listed in the "Inference input rules" section.

**Comment by gautammanak1** [comment]
## 👍 Summary

This PR refactors the query analyzer prompt system with a more comprehensive, structured prompt for ASI:One. The changes improve task detection accuracy, add better examples, and expand coverage for all modalities (text, image, audio, video, code). The fallback keyword-based system was also enhanced with additional keywords. Overall risk is **low** — this is a prompt engineering improvement with solid fallback handling.

## What I liked

- The prompt is now much more explicit about the output contract, which should reduce parsing errors from the LLM.
- Great decision to keep the keyword-based fallback — this makes the system resilient when the API is unavailable.
- The examples cover a wide range of realistic user queries across different modalities, which should help the model generalize better.
- Temperature is set to 0.0 for deterministic output — good choice for structured JSON responses.

## Security

- The API key is loaded from `config` module rather than hardcoded — good practice. ✅
- No hardcoded secrets detected in this file.

## Correctness / Logic

1. **Bare `except` clause** (line 142): The `try/except` block catches all exceptions with `except Exception:`. While this is intentional for fallback behavior, consider logging the exception for debugging:

   ```python
   except Exception as e:
       # Log the error for observability
       logger.warning(f"ASI:One analysis failed: {e}")
       return _fallback_analysis(user_text)
   ```

2. **JSON parsing assumptions** (lines 135-140): The code assumes the response structure always has `data["choices"][0]["message"]["content"]`. If the API returns an error response in a different format, this could raise `KeyError`. Consider validating the response structure before accessing nested keys.

3. **Task validation** (lines 142-144): The code validates that `result.get("task")` is in `TASK_MAP`, but doesn't validate other required fields like `search_query`, `inference_input`, or `needs_inference`. A malformed response could slip through.

## Performance

- The 60-second timeout is reasonable for LLM inference. ✅
- The fallback analysis is O(n*m) where n is the number of tasks and m is the average keywords per task — this is fine for the current scale (~20 tasks).
- Consider caching frequent user queries if this becomes a hot path.

## Code Quality

1. **Long lines**: The prompt string has several lines exceeding 120 characters (lines 68-79). While this is a prompt string and harder to split, consider using textwrap.dedent or implicit string concatenation for better readability:

   ```python
   ANALYZER_PROMPT = """\
   You are an expert HuggingFace model selector and query normalizer.
   
   Your job for ANY user request:
   1) Pick the single best HuggingFace pipeline task.
   2) Produce a concise model search query for HuggingFace model search.
   3) Produce the exact input payload text the selected model should receive.
   4) Decide whether inference should run now.
   ...
   """.strip()
   ```

2. **Missing type hints**: The `analyze_query` and `_fallback_analysis` functions return `dict` but could be more specific with a TypedDict or dataclass:

   ```python
   from typing import TypedDict
   
   class QueryAnalysis(TypedDict):
       task: str
       search_query: str
       inference_input: str
       needs_inference: bool
   
   async def analyze_query(user_text: str) -> QueryAnalysis:
       ...
   ```

3. **Magic string handling** (lines 135-140): The JSON cleaning logic strips backticks and the "json" prefix, but doesn't handle all possible markdown variations. Consider a more robust approach:

   ```python
   import re
   
   # Extract JSON from markdown code blocks
   json_match = re.search(r'```(?:json)?\s*({.*?})\s*```', content, re.DOTALL)
   if json_match:
       content = json_match.group(1)
   else:
       # Fallback to existing logic
       content = content.strip("`").strip()
       if content.startswith("json"):
           content = content[4:].strip()
   ```

## Tests

- No test files were changed in this PR. Consider adding tests for:
  - Valid JSON responses from ASI:One
  - Malformed JSON responses (should trigger fallback)
  - API errors (should trigger fallback)
  - Edge cases in keyword matching (e.g., "generate" matching both text and image generation)
  - Task validation with invalid task names

## Suggested Fixes

**Top priority:** Add logging to the exception handler for observability:

```python
import logging

logger = logging.getLogger(__name__)

# In analyze_query:
except Exception as e:
    logger.warning(f"ASI:One analysis failed: {e}", exc_info=True)
    return _fallback_analysis(user_text)
```

**Second priority:** Validate the complete response structure:

```python
# After parsing JSON
required_keys = {"task", "search_query", "inference_input", "needs_inference"}
if not required_keys.issubset(result.keys()):
    logger.warning(f"Missing required keys in response: {result}")
    return _fallback_analysis(user_text)

if result.get("task") not in TASK_MAP:
    result["task"] = DEFAULT_TASK
```

## Questions for the Author

1. Why was the `object-detection` task removed from the `task_keywords` dictionary in the fallback? It's present in the priority mapping hints but not in the keyword lists.

2. Have you considered adding a confidence score from the LLM response? This could help decide when to use the fallback instead of a low-confidence prediction.

3. The prompt mentions "code-generation" as a task, but the fallback maps it to "text-generation". Is this intentional? Should the prompt also use "text-generation" for consistency?

---

Overall, this is a solid improvement to the prompt system. The enhanced examples and clearer output contract should significantly improve query analysis accuracy. The suggestions above are minor and mostly about robustness and observability.
